### PR TITLE
Introduced "reset_state" method to reset all model states in one line

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
Introduced "reset_state" method to simplify the User API.

Before:
```
        for (auto&& state : infer_request.query_state()) {
            state.reset();
        }
```

After:
`       infer_request.reset_state()`

### Tickets:
 - *CVS-124193*

### CODE REVIEW

File: src/inference/include/openvino/runtime/infer_request.hpp [hunk 1]
```diff
@@ -315,6 +315,12 @@ class OPENVINO_RUNTIME_API InferRequest {
      */
     std::vector<VariableState> query_state();
 
+    /**
+     * @brief Resets all internal variable states for relevant infer request to a value specified as
+     * default for the corresponding `ReadValue` node
+     */
+    void reset_state();
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/infer_request.hpp:322]: let's also reuse this function here
https://github.com/openvinotoolkit/openvino/blob/b31aad78b0c509fcc3583ff8f7d6ac2968bdfc60/src/tests/functional/shared_test_classes/include/shared_test_classes/subgraph/stateful_model.hpp#L30-L32
to ensure it's used somewhere

File: src/inference/src/infer_request.cpp [hunk 2]
```diff
@@ -291,6 +291,12 @@ std::vector<VariableState> InferRequest::query_state() {
     return variable_states;
 }
 
+void InferRequest::reset_state(){OV_INFER_REQ_CALL_STATEMENT({
```
Comment (ilya-lavrenov) [inline src/inference/src/infer_request.cpp:294]: could you please fix indentation?
```suggestion
void InferRequest::reset_state(){
    OV_INFER_REQ_CALL_STATEMENT({
```
could you please fix indentation?

Comment (itikhono) [reply src/inference/src/infer_request.cpp:294]: clang-format did it :)

File: src/inference/include/openvino/runtime/infer_request.hpp (hunk 1 above)
Comment (itikhono) [reply src/inference/include/openvino/runtime/infer_request.hpp:322]: done

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**